tracing-opentelemetry/src/metrics.rs

```diff
@@ -53,10 +53,12 @@ impl Instruments {
             insert: impl FnOnce() -> T,
             update: impl FnOnce(&T),
         ) {
-            let lock = map.read().unwrap();
-            if let Some(metric) = lock.get(name) {
-                update(metric);
-                return;
+            {
+                let lock = map.read().unwrap();
+                if let Some(metric) = lock.get(name) {
+                    update(metric);
+                    return;
+                }
             }
```

Comment:
thanks!
yeah i just recently learned that when you reuse a variable name, rust doesn't drop the previous usage in case something else depends on that data